gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/filter/BundleFilter.java

```diff
@@ -53,4 +54,20 @@ public Bundle filter(String folderPath, FilterConfiguration filterConfiguration,
         return filteredBundle;
     }
 
+    public void validateImportedFolder(String folderPath, Bundle bundle) {
+        Boolean isValidImport = false;
+
+        // Verifies if node folder is found or if Root Node is found only if the base directory is specified
+        for (Folder folder : bundle.getFolders().values()) {
+            if (folderPath.endsWith(folder.getName()) || 
+                (folderPath.equals("/") && folder.getName().equals("Root Node"))) {
+                isValidImport = true;
+            }
+        }
+
+        if (!isValidImport) {
+            throw new RuntimeException("Unable to find specified folder \"" + folderPath + "\" in the Gateway connection folder path");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define and throw a dedicated exception instead of using a generic one. ([squid:S00112](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS00112))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=131&issues=AWcZobj9kEeD8QRklwaV&open=AWcZobj9kEeD8QRklwaV)

[ISSUE_KEY]: AWcZobj9kEeD8QRklwaV